connectors/stash/src/main/java/com/vmware/connectors/stash/StashController.java

```diff
@@ -0,0 +1,444 @@
+        final String description = stashResponse.read("$.description");
+        final boolean isPROpen = OPEN.equalsIgnoreCase(stashResponse.read("$.state"));
+
+        final Card.Builder card = new Card.Builder()
+                .setHeader(
+                        this.cardTextAccessor.getHeader(
+                                title,
+                                author),
+                        this.cardTextAccessor.getMessage(
+                                "stash.card.subtitle",
+                                description))
+                .setBody(makeCardBody(stashResponse, comments));
+
+        addCommentAction(card, routingPrefix, pullRequest);
+
+        // Add the following actions, only if the pull request state is open.
+        if (isPROpen) {
+            // Add approve action.
+            addPullRequestAction(card,
+                    routingPrefix,
+                    pullRequest,
+                    StashAction.APPROVE);
+
+            // Add decline action.
+            addPullRequestAction(card,
+                    routingPrefix,
+                    pullRequest,
+                    StashAction.DECLINE);
+
+            // Add merge action.
+            addPullRequestAction(card,
+                    routingPrefix,
+                    pullRequest,
+                    StashAction.MERGE);
+        }
+
+        return card.build();
+    }
+
+    private void addPullRequestAction(final Card.Builder card,
+                                      final String routingPrefix,
+                                      final StashPullRequest pullRequest,
+                                      final StashAction stashAction) {
+        card.addAction(
+                new CardAction.Builder()
+                    .setLabel(this.cardTextAccessor.getActionLabel(STASH_PREFIX + stashAction.getAction()))
+                    .setCompletedLabel(this.cardTextAccessor.getActionCompletedLabel(STASH_PREFIX + stashAction.getAction()))
+                    .setActionKey(CardActionKey.DIRECT)
+                    .setActionKey(buildActionUrl(routingPrefix, pullRequest, stashAction))
+                    .setType(HttpMethod.POST)
+                    .build()
+        );
+    }
+
+    private void addCommentAction(final Card.Builder card,
+                                  final String routingPrefix,
+                                  final StashPullRequest pullRequest) {
+        card.addAction(
+                new CardAction.Builder()
+                    .setLabel(this.cardTextAccessor.getActionLabel(STASH_COMMENTS))
+                    .setCompletedLabel(this.cardTextAccessor.getActionCompletedLabel(STASH_COMMENTS))
+                    .setActionKey(CardActionKey.USER_INPUT)
+                    .setUrl(buildActionUrl(routingPrefix, pullRequest, StashAction.COMMENTS))
+                    .addUserInputField(
+                            new CardActionInputField.Builder()
+                                .setId("Comment")
```

Comment:
This doesn't match your comment action code.  The client will pass in:

```
--------------------
Req URL
--------------------
http://jbard-m01.local/connectors/c5f0d317-9770-42c2-bf53-4102145a5ea9//api/v1/UFO/app-platform-server/252/comments

--------------------
Req Headers
--------------------
Content-Type : application/x-www-form-urlencoded
Accept-Language : en;q=0.9, *;q=0.5
X-Request-ID : 3F289EA3-B25E-4BE5-A9A4-B3DCBA815DDD
Authorization : [S] Bearer e*I
x-stash-authorization : [S] Basic am*=


--------------------
Req Body
--------------------
Comment=Testing
```

Because the id is "Comment", however, down below, you're expecting "comment" (from the COMMENT_PARAM_KEY).  Use the constant for both places.

I don't think your other actions have the same problem because they don't have user input.

Reply:
I have rectified this error now.. Below is the generated comment url
```
https://hero/connectors/stash/api/v1/UFO/app-platform-server/246/comments
```